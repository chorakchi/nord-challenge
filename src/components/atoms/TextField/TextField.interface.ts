import { InputHTMLAttributes } from 'react';


export interface TextFieldProps
  extends InputHTMLAttributes<HTMLInputElement> {
    label?: string;
    id: string;
    name?: string;
    Group?: Node;
}