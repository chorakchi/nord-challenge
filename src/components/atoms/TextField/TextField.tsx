import { TextFieldProps } from "./TextField.interface";
const TextField = ({ label, ...props }: TextFieldProps) => {
  return (
    <>
      <label htmlFor={props.id} className="sr-only">
        {label}
      </label>
      <input
        {...props}
        autoComplete={props.autoComplete || props.name}
        name={props.name || props.id}
        required
        className="appearance-none rounded-none relative block w-full px-3 py-2 border border-gray-300 placeholder-gray-500 text-gray-900 rounded-t-md focus:outline-none focus:ring-indigo-500 focus:border-indigo-500 focus:z-10 sm:text-sm"
        placeholder={props.placeholder || label}
      />
    </>
  );
};

// eslint-disable-next-line react/display-name
TextField.Group = ({ ...props }) => (
  <div {...props} className="rounded-md shadow-sm -space-y-px" />
);

export { TextField };
