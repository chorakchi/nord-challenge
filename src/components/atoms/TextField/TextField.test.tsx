import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { TextField } from "./TextField";

describe("TextField", () => {
  it("should render the TextField", () => {
    const subject = shallow(<TextField id="id" />);
    expect(subject).toHaveLength(1);
  });

  it("should render correctly and match to the Snapshot", () => {
    const subject = renderer.create(<TextField id="id" />).toJSON();
    expect(subject).toMatchSnapshot();
  });
});
