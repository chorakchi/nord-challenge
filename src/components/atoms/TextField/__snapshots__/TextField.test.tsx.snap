// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField should render correctly and match to the Snapshot 1`] = `
Array [
  <label
    className="sr-only"
    htmlFor="id"
  />,
  <input
    className="appearance-none rounded-none relative block w-full px-3 py-2 border border-gray-300 placeholder-gray-500 text-gray-900 rounded-t-md focus:outline-none focus:ring-indigo-500 focus:border-indigo-500 focus:z-10 sm:text-sm"
    id="id"
    name="id"
    required={true}
  />,
]
`;
